tools/distpartitioning/convert_partition.py

```diff
@@ -157,73 +157,108 @@ def create_dgl_object(graph_name, num_parts, \
+    reshuffle_nodes = th.arange(len(uniq_ids))
+    reshuffle_nodes = th.cat([reshuffle_nodes[inner_nodes.bool()],
+                              reshuffle_nodes[inner_nodes == 0]])
+
+    '''
+    Following procedure is used to map the part_local_src_id, part_local_dst_id to account for
+    reshuffling of nodes (to order localy owned nodes prior to non-local nodes in a partition)
+    1. Place side-by-side reshuffle_nodes and corresponding index value.▒
+        Basically (reshuffle_node[idx], idx) will form the key-value pair which we have to use
+        for mapping part_local_src_ids, and part_local_dst_ids.
+        Below we use column_stack to generate this mapping, call the resultant map as `node_map`
+
+    2. Sort the node_map array, so that the reshuffle_nodes column are sorted. This is needed because
+        `uniq_ids` are sorted as a result of np.unique operation.
+
+    3. Now, at this time the map is ready to map pre-node-reordering node-ids.▒
+
+    4. Now, concatenate [part_local_src_id, part_local_dst_id, np.arange(len(uniq_ids))] so that we have
+        the same sized array which is used to generate part_local_src_id and part_local_dst_id▒
+        which is the code above this comment section.
+
+    5. Now perform np.unique operation on the resultant list from step (4). This set of uniq_values
+        should follow the same order as `idx (uniq_ids)` above.▒
+
+    6. Now after this step, uniq_mapped_vals and uniq_ids should be the mapping between shuffle_global_nids
+        and part_local_nids
+
+    7. Now use the `node_map` array to map part_local_nids to their appropriate values to account for
+        node reordering in the `reshuffle_nodes` array.
+
+    Here is a  simple example to understand the above flow better.
+
+    part_local_nids = [0, 1, 2, 3, 4, 5]
+    part_local_src_ids = [0, 0, 0, 0, 2, 3, 4]
+    part_local_dst_ids = [1, 2, 3, 4, 4, 4, 5]
+
+    Assume that nodes {1, 5} are halo-nodes, which are not owned by this partition.
+
+    reshuffle_nodes = [0, 2, 3, 4, 1, 5]
+
+    A node_map, which maps node-ids from old to reshuffled order is as follows:
+    node_map = np.zeros((len(reshuffle_nodes,)))
+    node_map[reshuffle_nodes] = np.arange(len(reshuffle_nodes))
+
+    Using the above map, we have mapped part_local_src_ids and part_local_dst_ids as follows:
+    part_local_src_ids = [0, 0, 0, 0, 1, 2, 3]
+    part_local_dst_ids = [4, 1, 2, 3, 3, 3, 5]
+
+    In this graph above, note that nodes {0, 1, 2, 3} are inner_nodes and {4, 5} are NON-inner-nodes
+
+    Since the edge are re-ordered in any way, there is no reordering required for edge related data
+    during the DGL object creation.
+    '''
+    #create the mappings to generate mapped part_local_src_id and part_local_dst_id
+    #This map will map from unshuffled node-ids to reshuffled-node-ids (which are ordered to prioritize 
+    #locally owned nodes).
+    old2New = np.zeros((len(reshuffle_nodes,)))
```

Comment:
don't use capital characters. i think this is not very pythonic.

Reply:
renamed this to 'nid_map'